### ISSUE TITLE
Allow using ancestor classes in FinalRule whitelist parameter to whitelist their subclasses

### ISSUE THREAD

**Issue by rask** [ergebnis/phpstan-rules#96]
#### Steps required to reproduce the problem

1. Create a class as follows:

```php
class MyTestCase extends \PHPUnit\Framework\TestCase
{
    // ...
}
```

2. Add the following configuration to phpstan.neon:

```neon
services:
    -
        class: Localheinz\PHPStan\Rules\Classes\FinalRule
        arguments:
            allowAbstractClasses: true
            classesNotRequiredToBeAbstractOrFinal:
                - \PHPUnit\Framework\TestCase
        tags:
            - phpstan.rules.rule
```

3. 

#### Expected Result

* FinalRule does not complain about MyTestCase not being final or abstract

#### Actual Result

* `Class MyTestCase is neither abstract nor final.` is reported when running PHPStan.

---

Would it be sane to support parent classes and infer if a class inherits from them to allow skipping the final check? Or would a whole new option argument (e.g. `ancestorClassesNotRequiredToBeAbstractOrFinal`)?

Kinda defeats the purpose of this rule, but sometimes devs might have dependences that embrace the inheritance, and adding loads of exceptions into the config is not that feasible, and is error prone when it comes to teaching new devs about what that all is about and when to use it.
